src/SubscriberBase.h

```diff
@@ -83,6 +83,15 @@ class SubscriberBaseT : public Subscriber<T>,
       bool startExecutor = true)
       : ExecutorBase(executor, startExecutor), cancelled_(false) {}
 
+  virtual ~SubscriberBaseT() {
+    if (!cancelled_ && originalSubscription_) {
```

Comment:
Let me back up and describe my problem, and then maybe we can approach this a different way.

I have a class inheriting `SubscriberBaseT` which receives a `Subscription`. In the event that the _subscriber_ is destroyed, its subscription should cancel—otherwise, it might continue running, right? The problem is, `SubscriptionBase` is not cancellable when the `SubscriberBaseT` is being destructed, because it attempts to lock its `weak_ptr`, and it's already too late to do that.

Maybe an alternative way to solve this would be to make `SubscriptionBase` cancellable even if the parent `SubscriberBaseT` is no longer alive, but I'm not sure how best to achieve that.

What do you think?